app/src/main/java/com/mobileproto/hireddit/hireddit/SpeechFragment.java

```diff
@@ -0,0 +1,101 @@
+package com.mobileproto.hireddit.hireddit;
+
+import android.app.Fragment;
+import android.content.Intent;
+import android.os.Bundle;
+import android.speech.RecognizerIntent;
+import android.speech.SpeechRecognizer;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import java.util.ArrayList;
+import butterknife.Bind;
+import butterknife.ButterKnife;
+
+/**
+ * Speech Fragment: Holds functionality for requesting and recieving voice input
+ */
+public class SpeechFragment extends Fragment{
+    String TAG = "SpeechFragment Debug";
+    private View view;
+    private SpeechRecognizer sr;
+    private SpeechListener listener;
+    private Intent recognizerIntent;
+    private ArrayList voiceInput;
+    private boolean isListening = false;
+    @Bind(R.id.speech) Button speechButton;
+    @Bind(R.id.text) TextView textView;
+
+    public SpeechFragment() {
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        view = inflater.inflate(R.layout.fragment_speech, container, false);
+        ButterKnife.bind(this, view);
+
+        // *~Speech stuff~* //
+        listener = new SpeechListener(new SpeechCallback() {
+            @Override
+            public void callback(ArrayList voiceResult) {
+                voiceInput = voiceResult;
+                textView.setText(voiceInput.get(0).toString());
+                isListening = false;
+            }
+
+            @Override
+            public void errorCallback(int errorCode) {
+                if (errorCode == 7){
```

Comment:
Could you add comments or link to documentation about what these error codes are?